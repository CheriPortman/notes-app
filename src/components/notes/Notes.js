import React from 'react';
import PropTypes from 'prop-types';
import Note from './Note';

function Notes({ notes }) {
  const noteList = notes.map((note, index) => {
    return (
      <li key={index}>
        <Note note={note} />
      </li>
    );
  });

  return (
    <ul>
      {noteList}
    </ul>
  );
}

Notes.propTypes = {
  notes: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired
  })).isRequired
};

export default Notes;
