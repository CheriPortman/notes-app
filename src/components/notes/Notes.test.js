import React from 'react';
import { shallow } from 'enzyme';
import Notes from './Notes';

describe('Notes component', () => {
  it('renders the notes list', () => {
    const notes = [
      { title: 'to do', text: 'do this' },
      { title: 'another todo', text: 'do it again' }
    ];
    const wrapper = shallow(<Notes notes={notes} />);
    expect(wrapper).toMatchSnapshot();
  });
});
