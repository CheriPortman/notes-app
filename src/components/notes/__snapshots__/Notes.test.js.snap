// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes component renders the notes list 1`] = `
<styled.ul>
  <styled.li
    key="0"
  >
    <Note
      note={
        Object {
          "text": "do this",
          "title": "to do",
        }
      }
    />
  </styled.li>
  <styled.li
    key="1"
  >
    <Note
      note={
        Object {
          "text": "do it again",
          "title": "another todo",
        }
      }
    />
  </styled.li>
</styled.ul>
`;
